### PR TITLE
Migration 12 account_export_csv

### PR DESCRIPTION
**Pull request by bealdav**
Please could you have a look ? @vrenaville @yvaucher @sbidoul @pedrobaeza @guewen @legalsylvain

### CODE REVIEW

File: account_export_csv/wizard/account_export_csv_view.xml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                    <field name="journal_ids"/>
+                </div>
+               <separator string ="Report" colspan="4"/>
+                <group colspan="4">
+                    <field name="data" filename="export_filename"/>
+                </group>
+                <footer>
+                    <button name="action_manual_export_account" string="Trial Balance" type="object" icon="fa-cog" class="oe_highlight"/>
+                    <button name="action_manual_export_analytic" string="Analytic Balance (with accounts)" type="object" icon="fa-cog" class="oe_highlight"/>
+                    <button name="action_manual_export_journal_entries" string="Journal Entries" type="object" icon="fa-cog" class="oe_highlight"/>
```
Comment (bealdav) [inline account_export_csv/wizard/account_export_csv_view.xml]: oe_highlight class should be removed on every button.
too many highlight leads to confusion

File: account_export_csv/wizard/account_export_csv.py [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+        data = self.queue.getvalue()
+        # ... and reencode it into the target encoding as BytesIO
+        data = self.encoder.encode(data)
+        # write to the target stream
+        self.stream.write(data)
+        # empty queue with seek() instead of truncate()
+        # see https://stackoverflow.com/a/9729516
+        # also problems with seek() if next line is shorter than previous
+        # chars of previous line are kept in the new one
+        self.queue.flush()
```
Comment (yvaucher) [inline account_export_csv/wizard/account_export_csv.py]: We might want to just reinitialize the StringIO

According to https://stackoverflow.com/questions/4330812/how-do-i-clear-a-stringio-object
we could even simply create a new StringIO

```suggestion
        self.queue = StringIO()
```

**Review by yvaucher — APPROVED** [review]
LGTM just a proposition about StringIO emptying

File: account_export_csv/wizard/account_export_csv.py (hunk 2 above)
Comment (bealdav) [reply account_export_csv/wizard/account_export_csv.py]: Done

**Comment by bealdav** [comment]
please @guewen @vrenaville @sbidoul could you make a small review ? Thanks

**Comment by yvaucher** [comment]
ping for reviews, need one more.

File: account_export_csv/wizard/account_export_csv.py [hunk 3]
```diff
@@ -65,12 +59,17 @@ class AccountCSVExport(models.TransientModel):
...
+        comodel_name='account.journal', string='Journals',
+        default=lambda s: s._get_journal_default(),
         help='If empty, use all journals, only used for journal entries')
     export_filename = fields.Char(
         string='Export CSV Filename', size=128, default='account_export.csv')
 
+    @api.model
+    def _get_journal_default(self):
+        """ Implements your own default """
+        return False
```
Comment (sbidoul) [inline account_export_csv/wizard/account_export_csv.py:74]: Nit: Is it really useful to have this method, compared to letting people override with Odoo standard mechanisms?

Comment (sbidoul) [reply account_export_csv/wizard/account_export_csv.py:74]: Nit: Also for company, is it not better to use `_company_default_get`?

**Review by sbidoul — APPROVED** [review]
At first glance the AccountingWriter can probably be written in a much simpler way (using TextIOWrapper?). But I suppose it's not relevant for a migration. Otherwise LGTM, thanks!

**Comment by bealdav** [comment]
please could you consider merge this PR ?

Thanks

**Review by alexis-via — APPROVED** [review]

**Pull request merged**